### PR TITLE
Update service/client request/response API error returns

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Comment by Lobotuerk** [comment]
CI job up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12455/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7422/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10166/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12374/)

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -3685,10 +3685,17 @@ extern "C" rmw_ret_t rmw_send_request(
...
-  RET_NULL(sequence_id);
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    client handle,
+    client->implementation_identifier, eclipse_cyclonedds_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(ros_request, RMW_RET_INVALID_ARGUMENT);
+
```
Comment (hidmic) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:3696]: @Lobotuerk nit: remove the above three blank lines.

**Review by hidmic — APPROVED** [review]

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 1 above)
Comment (Lobotuerk) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Updated on 616b200

**Review by hidmic — APPROVED** [review]
LGTM pending green CI including tests.

**Comment by Lobotuerk** [comment]
CI up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12502/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7470/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10214/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12422/)

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 2]
```diff
@@ -3685,10 +3694,14 @@ extern "C" rmw_ret_t rmw_send_request(
   int64_t * sequence_id)
 {
   static std::atomic_uint next_request_id;
-  RET_NULL(client);
-  RET_WRONG_IMPLID(client);
-  RET_NULL(ros_request);
-  RET_NULL(sequence_id);
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    client handle,
```
Comment (Blast545) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:3699]: Was this left purposely ?

**Review by Blast545 — APPROVED** [review]

**Comment by Lobotuerk** [comment]
CI after reviews, up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12509/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7477/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10221/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12429/)

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 2 above)
Comment (Lobotuerk) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Changed on 0113709

**Comment by Lobotuerk** [comment]
@ros-pull-request-builder retest this please.

**Comment by hidmic** [comment]
Github Actions keep complaining about internal errors in MacOS jobs. Consider our CI is green and all other jobs are passing, I'll move forward with this PR.

**Pull request merged**